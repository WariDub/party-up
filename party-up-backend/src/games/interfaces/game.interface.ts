export interface Game {
    id: number;
    cover: { [key: string]: string };
    genres: { [key: string]: string };
    name: string;
    rating: number;
}
