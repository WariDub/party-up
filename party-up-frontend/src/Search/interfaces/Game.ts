export interface Game {
  id: string;
  cover: GameCover;
  genres: { [key: string]: string };
  name: string;
  rating: number;
}

interface GameCover {
  url: string;
}
