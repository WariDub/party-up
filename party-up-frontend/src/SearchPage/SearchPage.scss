html,
body {
  padding: 0;
  margin: 0;
  color: #292929;
  background-color: green;
}


