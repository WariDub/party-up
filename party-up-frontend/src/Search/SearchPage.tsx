import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import GamesList from './components/GamesList';
import NavBar from './components/NavBar';
import { Game } from './interfaces/Game';

export interface SearchPageProps extends RouteComponentProps {}

export interface SearchPageState {
  results: Game[] | null;
}

const SearchPage = class extends React.PureComponent<SearchPageProps, SearchPageState> {
  constructor(props: SearchPageProps) {
    super(props);

    this.state = {
      results: [],
    };
  }

  render(): JSX.Element {
    const { history, location, match } = this.props;
    const { results } = this.state;

    return (
      <>
        <NavBar
          history={history}
          location={location}
          match={match}
          showSearchBar
          onQueryResult={this.onQueryResult}
        />
        <GamesList history={history} location={location} match={match} results={results} />
      </>
    );
  }

  onQueryResult = (data: Game[] | null): void => {
    this.setState({ results: data });
  };
};

export default SearchPage;
