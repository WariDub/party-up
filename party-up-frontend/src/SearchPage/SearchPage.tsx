import React, {Component, Fragment} from 'react';

import './SearchPage.scss';
import SearchBar from './SearchBar/SearchBar';
import ToolBar from './ToolBar/ToolBar';



class SearchPage extends Component{
    render(){
        return(
            <Fragment>
                <ToolBar/>
                <SearchBar/>
            </Fragment>           
        )
    }
}

export default SearchPage;