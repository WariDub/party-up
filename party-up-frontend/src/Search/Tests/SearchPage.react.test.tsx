import { shallow } from 'enzyme' 
import renderer from 'react-test-renderer';
import SearchPage from '../SearchPage';
import { Game } from '../../Search/interfaces/Game';
import { createBrowserHistory } from 'history'
import { matchPath } from 'react-router';

describe('Search Page', () => {
    test('It should render without crashing', () => {
        let history = createBrowserHistory();
        history.push('/');

        let location = history.location;

        let match = matchPath('/ProfilePage', {
            path: '/ProfilePage',
            exact: true,
            strict: false,
        });
        
        let component = shallow(<SearchPage history = { history } location = { location } match = { match } />);

        expect(component.length).toBe(1);
    });

    test('It should render without crashing', () => {
        let history = createBrowserHistory();
        history.push('/');

        let location = history.location;

        let match = matchPath('/ProfilePage', {
            path: '/ProfilePage',
            exact: true,
            strict: false,
        });
        
        let component = renderer.create(<SearchPage history = { history } location = { location } match = { match } />).toJSON();

        expect(component).toMatchSnapshot();
    });
});