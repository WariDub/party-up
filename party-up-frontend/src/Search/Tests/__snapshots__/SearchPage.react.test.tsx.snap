// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page It should render without crashing 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          PartyUp
        </a>
      </h6>
      <div
        className="MuiBox-root MuiBox-root-1"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
          onKeyDown={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
            data-shrink={false}
          >
            Enter game name
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-justify-xs-flex-end"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          id="ProfileButton"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Profile
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-justify-xs-flex-end"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Logout
          </span>
        </button>
      </div>
    </div>
  </header>,
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <ul
      className="MuiGridList-root"
      style={
        Object {
          "margin": -4,
        }
      }
    />
  </div>,
]
`;
